aspnetcore/migration/22-to-30.md

```diff
@@ -607,3 +607,57 @@ If you're using features of `Newtonsoft.Json` that aren't supported in `System.T
 ## Opt in to runtime compilation
 
 In 3.0, runtime compilation is an opt-in scenario. To enable runtime compilation, see https://docs.microsoft.com/aspnet/core/mvc/views/view-compilation?view=aspnetcore-3.0#runtime-compilation.
+
+## Blazor event handlers and bind attributes
+
+In ASP.NET Core Preview 9 or later, the set of bindings and event handlers available for HTML elements are:
+
+* Defined in the `Microsoft.AspNetCore.Components.Web` assembly (`Microsoft.AspNetCore.Components.Web` NuGet package).
+* Found in the `Microsoft.AspNetCore.Components.Web` namespace.
+
+### Migration
+
+**Blazor WebAssembly and Razor Class Libraries**
+
+When migrating an app to ASP.NET Core Preview 9 or later:
+
+1. Add a package reference to the `Microsoft.AspNetCore.Components.Web` package in the project file.
+1. Open the app's root *_Imports.razor* file and add `@using Microsoft.AspNetCore.Components.Web`.
+
+**Blazor server-side**
+
+When migrating an app to ASP.NET Core Preview 9 or later, open the app's root *_Imports.razor* file and add `@using Microsoft.AspNetCore.Components.Web`.
+
+> [!NOTE]
+> A Blazor server-side app receives a `Microsoft.AspNetCore.Components.Web` package reference implicitly, so manually adding the package reference to the app's project file isn't required.
+
+### Troubleshoot
+
+**Syntax highlighting in Visual Studio**
+
+If `@bind` or `@on...` attributes are colorized as a normal HTML attributes (shown in the following image), the `@using` statement is missing:
+
+![Index.razor component CSHTML markup in Visual Studio showing HTML colorized unemphasized font for bind and onclick attributes.](https://user-images.githubusercontent.com/1430011/64197856-f1679300-ce3b-11e9-83b7-71ffa6d68fec.png)
+
+With the correct package reference and `@using` statement, event handlers and data binding attributes, such as `@onclick` and `@bind`, appear in a bold font and colorized as Razor syntax as shown in the following images when using Visual Studio:
+
+![Index.razor component CSHTML markup in Visual Studio showing Razor colorized bold font for bind and onclick attributes.](https://user-images.githubusercontent.com/1430011/64197727-a3528f80-ce3b-11e9-8cc4-fe34a858eb49.png)
+
+**Missing Microsoft.AspNetCore.Components.Web package reference**
+
+The following error appears if the reference to the `Microsoft.AspNetCore.Components.Web` package is missing from the project:
+
+```
+CS0169 The field 'Index.text' is never used
+CS0428 Cannot convert method group 'Submit' to non-delegate type 'object'. Did you intend to invoke the method?
```

Comment:
This isn't really right. These errors can occur due to either issue. Depending on the exact code you wrote, you may or may not see these issues.